opentelemetry/src/logs/record.rs

```diff
@@ -26,7 +26,11 @@ pub trait LogRecord {
     fn set_body(&mut self, body: AnyValue);
 
     /// Adds multiple attributes.
-    fn add_attributes(&mut self, attributes: Vec<(Key, AnyValue)>);
+    fn add_attributes<I, K, V>(&mut self, attributes: I)
```

Comment:
would be good to leave a comment explaining that this API is designed to take ownership, unlike Metric APIs which just need slice/ref only to attributes.